Bio/Seq.py

```diff
@@ -1280,6 +1281,53 @@ def ungap(self, gap=None):
             raise ValueError("Unexpected gap character, {0!r}".format(gap))
         return Seq(str(self).replace(gap, ""), alpha)
 
+    def join(self, other):
+        """Return a merge of the sequences in other, spaced by the sequence from self.
+
+        Accepts all Seq objects and Strings as objects to be concatenated with the spacer
+
+        >>> concatenated = Seq('NNNNN').join([Seq("AAA"), Seq("TTT"), Seq("PPP")])
+        >>> concatenated
+        Seq('AAANNNNNTTTNNNNNPPP')
+
+        Throws error if other is not an iterable and if objects inside of the iterable
+        are not Seq or String objects
+        """
+        if not isinstance(other, collections.Iterable):  # doesn't detect single strings
+            raise ValueError("Input must be an iterable")
+        if isinstance(other, basestring):
+            raise ValueError("Input must be an iterable")
+        from Bio.SeqRecord import SeqRecord  # Lazy to avoid circular imports
+        temp_data = ""
+        # if the spacer is empty it will initially default none
+        if self._data == "":
+            a = None
+        else:
+            a = self.alphabet
+        for c in other:
+            if isinstance(c, SeqRecord):
+                raise TypeError("Iterable cannot contain SeqRecords")
+            elif hasattr(c, "alphabet"):
+                if a is None:  # if spacer is empty alphabet defaults to type of the first sequence
+                    a = c.alphabet
+                else:
+                    if not Alphabet._check_type_compatible([a, c.alphabet]):
```

Comment:
I found that all these alphabet checks are intrinsicely slow. Also, I would think that often the alphabets are identical. So you can speed this up considerably if you only do the following checks if the alphabets are not identical:
```python
if a != c.alphabet:
    if not Alphabet._ ....
        ...
```
In a test run with 1000 'AAA' seq objects, this check made the method nearly 9 times faster.